judge-d-server/src/main/java/com/hltech/judged/server/interfaces/rest/contracts/ContractsController.java

```diff
@@ -99,6 +100,20 @@ public ServiceContractsDto getContracts(@PathVariable(name = "serviceName") Stri
         return mapper.toDto(this.serviceContractsRepository.findOne(new ServiceId(serviceName, version)).orElseThrow(ResourceNotFoundException::new));
     }
 
+    @GetMapping(value = "services/{serviceName}/versions/{version:.+}/capabilities/{protocol}", produces = MediaType.APPLICATION_JSON_VALUE)
+    @CrossOrigin()
+    @ApiOperation(value = "Get capabilities of a version of a service for a protocol", nickname = "get capabilities by protocol")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Success", response = ServiceContractsDto.class),
+        @ApiResponse(code = 400, message = "Bad Request"),
+        @ApiResponse(code = 500, message = "Failure")})
+    public String getCapabilities( @PathVariable String serviceName, @PathVariable String version, @PathVariable(name = "protocol") String protocol) {
```

Comment:
You should return contracts with their mime type as they are registered

Reply:
Endpoint modified to return mime type of a contract.